src/providers/tfc/runProvider.ts

```diff
@@ -47,11 +51,69 @@ export class RunTreeDataProvider implements vscode.TreeDataProvider<vscode.TreeI
         this.reporter.sendTelemetryEvent('tfc-runs-viewInBrowser');
         const runURL = `${TerraformCloudWebUrl}/${orgName}/workspaces/${run.workspace.attributes.name}/runs/${run.id}`;
 
+        vscode.env.openExternal(vscode.Uri.parse(runURL));
+      }),
+      vscode.commands.registerCommand('terraform.cloud.run.plan.downloadLog', async (run: RunTreeItem) => {
+        // get plan by id from run
+        const plan = await apiClient.getPlan({
+          params: {
+            plan_id: run.planId ?? '',
```

Comment:
I include plans when getting all in order to detect whether I show the plan button or not.

I re-get the plan when pressed because according to https://developer.hashicorp.com/terraform/cloud-docs/api-docs#blob-storage-authentication, log-read-url is regenerated every 25 hours. If I used the one from listing all runs, and the user clicked at some point in the future, it may be invalid. Or did I read that wrong?